code/modules/FTL/Combat/phase_cannon.dm

```diff
@@ -64,50 +64,53 @@
 /obj/machinery/power/shipweapon/proc/can_fire()
 	if(state != 2)
 		return 0
-	return cell.charge >= 500
+	return power_charge == attack_type.required_charge
 
 /obj/machinery/power/shipweapon/proc/attempt_fire(var/datum/component/target_component)
 	if(!can_fire())
 		return 0
-	cell.use(500)
-
-	var/obj/item/projectile/ship_projectile/A = new projectile_type(src.loc)
-
-	A.setDir(src.dir)
-	playsound(src.loc, projectile_sound, 50, 1)
-	for(var/obj/machinery/computer/ftl_weapons/C in world)
-		if(!istype(get_area(C), /area/shuttle/ftl))
-			continue
-		if(!(src in C.laser_weapons))
-			continue
-		playsound(C, projectile_sound, 50, 1)
-
-	if(prob(35))
-		var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
-		s.set_up(5, 1, src)
-		s.start()
-
-	switch(dir)
-		if(NORTH)
-			A.yo = 20
-			A.xo = 0
-		if(EAST)
-			A.yo = 0
-			A.xo = 20
-		if(WEST)
-			A.yo = 0
-			A.xo = -20
-		else // Any other
-			A.yo = -20
-			A.xo = 0
+	power_charge = 0
+
+	for(var/i = 1 to attack_type.shot_amount)
+		var/obj/item/projectile/ship_projectile/A = new attack_type.projectile_type(src.loc)
+
+		A.setDir(EAST)
+		A.pixel_x = 32
+		playsound(src.loc, attack_type.projectile_sound, 50, 1)
+		for(var/obj/machinery/computer/ftl_weapons/C in world)
+			if(!istype(get_area(C), /area/shuttle/ftl))
+				continue
+			if(!(src in C.laser_weapons))
+				continue
+			playsound(C, attack_type.projectile_sound, 50, 1)
+
+		if(prob(35))
+			var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+			s.set_up(5, 1, src)
+			s.start()
+
+		A.yo = 0
+		A.xo = 20
 		A.starting = loc
-	A.fire()
-	A.target = target_component
-	update_icon()
 
+		A.fire()
+		A.target = target_component
+		sleep(5)
```

Comment:
I'm iffy about this whole loop tbh. It should probably be a seperate proc alled by addtimer with delays between each and a CHECK_TICK